api/tests/unit/features/test_unit_features_features_service.py

```diff
@@ -1,35 +1,184 @@
+from typing import TYPE_CHECKING
+from unittest.mock import ANY
+
+import pytest
+
+from edge_api.identities.models import EdgeIdentity
 from environments.identities.models import Identity
-from features.features_service import get_overrides_data
+from features.features_service import (
+    get_core_overrides_data,
+    get_edge_overrides_data,
+    get_overrides_data,
+)
 from features.models import Feature, FeatureSegment, FeatureState
+from projects.models import IdentityOverridesV2MigrationStatus, Project
+from util.mappers.engine import (
+    map_feature_state_to_engine,
+    map_identity_to_engine,
+)
 
+if TYPE_CHECKING:
+    from pytest_mock import MockerFixture
 
-def test_feature_get_overrides_data(
-    feature,
-    environment,
-    identity,
-    segment,
-    feature_segment,
-    identity_featurestate,
-    segment_featurestate,
-):
-    # Given
-    # we create some other features with overrides to ensure we're only getting data
-    # for each individual feature
-    feature_2 = Feature.objects.create(project=feature.project, name="feature_2")
-    FeatureState.objects.create(
-        feature=feature_2, environment=environment, identity=identity
+    from environments.dynamodb.dynamodb_wrapper import (
+        DynamoEnvironmentV2Wrapper,
+        DynamoIdentityWrapper,
     )
+    from environments.models import Environment
+    from segments.models import Segment
+
 
-    feature_3 = Feature.objects.create(project=feature.project, name="feature_3")
-    feature_segment_for_feature_3 = FeatureSegment.objects.create(
-        feature=feature_3, segment=segment, environment=environment
+@pytest.fixture
+def distinct_segment_featurestate(
```

Comment:
It's not clear to me what this is 'distinct' from?

Reply:
Ah, I think it's a feature that is overridden by a segment only, not an identity? Is this necessary, can we use lazy_fixture below perhaps to parametrise the test?

Reply:
> It's not clear to me what this is 'distinct' from?

From the `segment_featurestate` fixture, which is for the same feature as `identity_featurestate`. I'll welcome better naming propositions. 

I don't think we need to parametrise the tests as they handle all the cases at once.